### PR TITLE
Rewards Component. Gets data from lnbits Marketplace

### PR DESCRIPTION
**Pull request by NataliaMilgevskaya**
please, pay attention: lnbits market has image size limitation: 252b, so the images on the screen with bad quality

### CODE REVIEW

File: tabs/src/Home.tsx [hunk 1]
```diff
@@ -1,10 +1,11 @@
 import React from 'react';
 import FeedComponent from './components/FeedComponent';
+import RewardsComponent from "./components/RewardsComponent";
 
 const Home: React.FC = () => {
   return (
     <div>
-      <FeedComponent />
+            <RewardsComponent />
```
Comment (BenGWeeks) [inline tabs/src/Home.tsx:8]: This should go on a new `src/Rewards.tsx` page.

File: tabs/src/global.d.ts
Comment (BenGWeeks) [inline tabs/src/global.d.ts:1]: You might need to get latest here.

File: tabs/src/interfaces/DataModel.ts
Comment (BenGWeeks) [inline tabs/src/interfaces/DataModel.ts:1]: This should go in `tabs/src/global.d.ts` (remove carriage returns).

File: tabs/src/services/lnbitsServiceLocal.ts [hunk 2]
```diff
@@ -500,6 +500,32 @@ const getWalletIdByUserId = async (adminKey: string, userId: string) => {
...
+      throw new Error(`Error getting products (status: ${response.status})`);
+    }
+
+    const data = await response.json();
+    console.log('Products:', data);
+
+    return data;  // This will return the list of products
+  } catch (error) {
+    console.error('Error getting products:', error);
+    return null;
```
Comment (BenGWeeks) [inline tabs/src/services/lnbitsServiceLocal.ts:525]: `throw error`
(need to update others to be that too).

File: tabs/src/services/lnbitsServiceLocal.ts [hunk 3]
```diff
@@ -500,6 +500,32 @@ const getWalletIdByUserId = async (adminKey: string, userId: string) => {
   }
 };
 
+const getRewards = async (apiKey: string) => {
```
Comment (BenGWeeks) [inline tabs/src/services/lnbitsServiceLocal.ts:503]: Should we change `apiKey` to either `adminKey`? Not sure what rewards uses as the X-Api-Key.

File: tabs/src/services/lnbitsServiceLocal.ts [hunk 4]
```diff
@@ -500,6 +500,32 @@ const getWalletIdByUserId = async (adminKey: string, userId: string) => {
...
+  console.log('Getting products ...');
+  try {
+    const response = await fetch('https://demo.lnbits.com/market/api/v1/products?all_stalls=false', {
+      method: 'GET',
+      headers: {
+        'Content-Type': 'application/json',
+        // 'X-Api-Key': apiKey,  // Pass the API key here
```
Comment (BenGWeeks) [inline tabs/src/services/lnbitsServiceLocal.ts:510]: Remove duplicate comments line out.

File: tabs/src/services/lnbitsServiceLocal.ts (hunk 3 above)
Comment (BenGWeeks) [inline tabs/src/services/lnbitsServiceLocal.ts:503]: Return a Promise (good practice to return types).

File: tabs/src/services/lnbitsServiceLocal.ts [hunk 5]
```diff
@@ -500,6 +500,32 @@ const getWalletIdByUserId = async (adminKey: string, userId: string) => {
...
+    });
+
+    if (!response.ok) {
+      throw new Error(`Error getting products (status: ${response.status})`);
+    }
+
+    const data = await response.json();
+    console.log('Products:', data);
+
+    return data;  // This will return the list of products
```
Comment (BenGWeeks) [inline tabs/src/services/lnbitsServiceLocal.ts:522]: Return an array of `Rewards[]`.

File: tabs/src/types/global.d.ts [hunk 6]
```diff
@@ -28,7 +28,7 @@ interface Wallet {
 
 type WalletType = 'Sending' | 'Receiving';
 
-declare module '*.svg' {
-  const content: any;
-  export default content;
+declare module '*.module.css' {
```
Comment (BenGWeeks) [inline tabs/src/types/global.d.ts:31]: Wondering if this should go in here or not?

**Review by BenGWeeks — CHANGES_REQUESTED** [review]

File: tabs/src/Home.tsx (hunk 1 above)
Comment (BenGWeeks) [reply tabs/src/Home.tsx:8]: And add the Rewards page to the FooterComponent.

**Comment by BenGWeeks** [comment]
Would be good to include a screenshot in your PR :-)

Suggest making changes in a few more commits so I can see history of changes.

File: tabs/src/Home.tsx (hunk 1 above)
Comment (NataliaMilgevskaya) [reply tabs/src/Home.tsx:8]: > This should go on a new `src/Rewards.tsx` page.

resolved

File: tabs/src/interfaces/DataModel.ts
Comment (NataliaMilgevskaya) [reply tabs/src/interfaces/DataModel.ts:1]: > This should go in `tabs/src/global.d.ts` (remove carriage returns).

resolved

File: tabs/src/services/lnbitsServiceLocal.ts (hunk 4 above)
Comment (NataliaMilgevskaya) [reply tabs/src/services/lnbitsServiceLocal.ts:510]: resolved

File: tabs/src/types/global.d.ts (hunk 6 above)
Comment (NataliaMilgevskaya) [reply tabs/src/types/global.d.ts:31]: resolved

**Comment by NataliaMilgevskaya** [comment]
> Would be good to include a screenshot in your PR :-)
> 


> Suggest making changes in a few more commits so I can see history of changes.

File: tabs/src/services/lnbitsServiceLocal.ts [hunk 7]
```diff
@@ -500,6 +500,32 @@ const getWalletIdByUserId = async (adminKey: string, userId: string) => {
   }
 };
 
+const getRewards = async (apiKey: string) => {
+  console.log('Getting products ...');
+  try {
+    const response = await fetch('https://demo.lnbits.com/market/api/v1/products?all_stalls=false', {
```
Comment (BenGWeeks) [inline tabs/src/services/lnbitsServiceLocal.ts:506]: Check how the other methods work here (environment variable for the Url) with an entry in setupProxy.js for routing to stop CORS errors. (Ref: https://demo.lnbits.com/market/api/v1/products)

**Review by BenGWeeks — CHANGES_REQUESTED** [review]

**Comment by BenGWeeks** [comment]
Minor tweaks, but we should update this to use the `nostrmarket` extension. I found the repo for `marketplace` here:

[lnbits/market: market extension, inspired by nostrmarket (github.com)](https://github.com/lnbits/market?tab=readme-ov-file)

But see:

image

Makes our mind up, and probably why not available on my.lnbits.com anymore.

**Comment by BenGWeeks** [comment]
NB Images should be links. Should use a CDN for the images.

**Comment by NataliaMilgevskaya** [comment]
> Should use a CDN for the images.

we used https://postimg.cc/ as CDN. Is it OK? Super User can select other variants.


> NB Images should be links. 
should it open the site link?  the same functionality if we click on 'Product details'

File: tabs/src/services/lnbitsServiceLocal.ts (hunk 7 above)
Comment (BenGWeeks) [reply tabs/src/services/lnbitsServiceLocal.ts]: Looks like you still have the LNBits instant hardcoded. See other functions don't use this.

**Comment by BenGWeeks** [comment]
See review comment / requested change above.

File: tabs/src/services/lnbitsServiceLocal.ts (hunk 7 above)
Comment (NataliaMilgevskaya) [reply tabs/src/services/lnbitsServiceLocal.ts]: getRewards is not actual function. I've removed it and replaced with getNostrRewards

image

Comment (BenGWeeks) [reply tabs/src/services/lnbitsServiceLocal.ts]: And using Promises / Types xxx

**Comment by BenGWeeks** [comment]
Looks like it's ready for testing :-)

**Pull request merged**